unity-client/Assets/Scripts/MainScripts/DCL/Components/Avatar/AvatarVisibility.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using UnityEngine;
+using UnityEngine.Assertions;
+
+namespace DCL
+{
+    public class AvatarVisibility : MonoBehaviour
+    {
+
+        public AvatarRenderer renderer;
+        [Tooltip("Game objects that should be hidden/shown together with the renderer")]
+        public GameObject[] gameObjectsToToggle;
+        // When this property is true, then the visibility can't be modified
+        private Boolean locked = false;
+        private Boolean? isVisible;
+
+        public void SetVisibility(Boolean visibility)
+        {
+            if (locked == false && isVisible != visibility)
```

Comment:
This lock logic works fine here, but may produce some issues with avatars that are mid-loading and enter the area, as the renderers `enabled` state is enabled in `WearableController.SetAssetRenderersEnabled` (and potentially elsewhere). 

I think we should hide the avatars in a way that doesn't overlap the `renderers.enabled` value or at the very least wrap all the state changes with this approach.

Reply:
I think it's probably better if we move this `SetVisibility` (and its locking) into the `AvatarRenderer` itself, I understand that the behaviour wouldn't be the same for other avatars and the player avatar, but the `CharacterController` has its own `AvatarRenderer` (in a child GameObject), can't we use that in the same manner as we use the AvatarRenderer of the other players?

As Brian says we could unify the logic there, and instead of toggling the gameobject in `AvatarRenderer.SetVisibility()` we can use the `WearableController.SetAssetRenderersEnabled()` of each item in `AvatarRenderer.wearableControllers`

Reply:
I think it's probably better if we move this `SetVisibility` (and its locking) into the `AvatarRenderer` itself, I understand that the behaviour wouldn't be the same for other avatars and the player avatar, but the `CharacterController` has its own `AvatarRenderer` (in a child GameObject), can't we use that in the same manner as we use the AvatarRenderer of the other players?

As Brian says we could unify the logic there, and instead of toggling the gameobject in `AvatarRenderer.SetVisibility()` we can use the `WearableController.SetAssetRenderersEnabled()` of each item in `AvatarRenderer.wearableControllers`